src/common/runnable/run-many-sync-for.ts

```diff
@@ -19,7 +19,7 @@ type RunSync<Param> = (parameter: Param) => void;
 export type RunManySync = <Param>(injectionToken: InjectionToken<RunnableSync<Param>, void>) => RunSync<Param>;
 
 function runCompositeRunnableSyncs<Param>(param: Param, composite: Composite<RunnableSync<Param>>) {
-  assert(!((composite.value.run(param) as any) instanceof Promise), "Cannot be an async function for runnable sync");
+  assert(!((composite.value.run(param) as unknown) instanceof Promise), `Cannot be an async function for runnable sync`);
```

Comment:
I wonder if there's a way to enforce "no async" with typing instead of at runtime. I do remember it being a bit tricky with TypeScript, but still.

Reply:
Not if we want implicit returns to be allowed. Typescript treats the `void` return type as special allowing all functions to coerce to it.

We could change it to be `undefined` but then the functions would need an explicit `return undefined` to be typed correctly.